servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/AbstractContentCodec.java

```diff
@@ -30,20 +31,20 @@ public final CharSequence name() {
     }
 
     @Override
-    public boolean equals(final Object o) {
+    public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof AbstractContentCodec)) {
             return false;
         }
         final AbstractContentCodec that = (AbstractContentCodec) o;
-        return name.equals(that.name);
+        return contentEquals(name, that.name);
```

Comment:
Great catch! I also found in RFC that all content-coding values are case-insensitive, see https://tools.ietf.org/html/rfc7231#section-3.1.2.1. That means that the `hashCode` impl here is correct 😄 

Let's merge this PR as is (don't want to mix 2 issues in one PR) and I've filed #1385 for the follow-up if you are interested.